lesson-003-web-server-graceful-shutdown/main.go

```diff
@@ -0,0 +1,78 @@
+package main
+
+import (
+	"context"
+	"fmt"
+	"log"
+	"net/http"
+	"os"
+	"os/signal"
+	"syscall"
+	"time"
+)
+
+// listenAddr is the address where our web server will be listening for
+// requests.
+const listenAddr = "0.0.0.0:8080"
+
+// homeHandler takes a response writer to build a response for the given
+// request.
+// This http handler will greet you with the hostname of the machine where this
+// app is running.
+func homeHandler(w http.ResponseWriter, r *http.Request) {
```

Comment:
For this example, it may be nice to add a "delay" to this function to showcase that a graceful shutdown is letting the server handle connections it already created instead of terminating the whole thing at once.

Reply:
Agree, in fact I had a timeout when I was testing it locally 😅